src/test/projects/include-exclude/pom.xml

```diff
@@ -0,0 +1,51 @@
+<!--
+
+    Copyright 2021 the original author or authors.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+         http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>org.apache.maven.caching.test.include-exclude</groupId>
+    <artifactId>include-exclude</artifactId>
+    <version>0.0.1-SNAPSHOT</version>
+    <packaging>jar</packaging>
+
+    <properties>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
+        <!-- Testing the project properties to include / exclude -->
+
+        <!-- one more file and one more folder to scan -->
+        <maven.build.cache.input.extraFile>extraFile.txt</maven.build.cache.input.extraFile>
+        <maven.build.cache.input.extraFolder>second_folder_outside_src</maven.build.cache.input.extraFolder>
+        <!-- Exclude directory in an extra included folder  -->
+        <maven.build.cache.exclude.subfolder>**/excluded_subfolder/**</maven.build.cache.exclude.subfolder>
```

Comment:
Please notice that the documentation describes the parameters as follows:
```
Paths to exclude from source code search. Relative paths calculated from the current project/module root
```
Please notice the documentation declares this not as a pattern, a glob, or an ant expression. We need to update the documentation to reflect the support for globs and how they are applied. The key issue is whether the base dir is prepended, with or without leading slashes. To make it deterministic, we might follow this approach:
* If the pattern starts with `/`, it is considered an absolute path expression
* If not, any path expression is prepended by the absolute project dir path
* The content of the properties will be interpreted as a glob expression.